src/main/scala/firrtl/Emitter.scala

```diff
@@ -96,15 +93,18 @@ sealed abstract class FirrtlEmitter(form: CircuitForm) extends Transform with Em
   }
 
   override def execute(state: CircuitState): CircuitState = {
-    val newAnnos = state.annotations.flatMap {
-      case EmitCircuitAnnotation(_) =>
-        Seq(EmittedFirrtlCircuitAnnotation.apply(
-              EmittedFirrtlCircuit(state.circuit.main, state.circuit.serialize)))
-      case EmitAllModulesAnnotation(_) =>
-        emitAllModules(state.circuit) map (EmittedFirrtlModuleAnnotation(_))
-      case _ => Seq()
+    val (e, other) = state.annotations.partition{
```

Comment:
A comment on what is going on with annotations here would be helpful

Reply:
I've added comments for the three major steps going on here.